src/pykoala/corrections/sky.py

```diff
@@ -71,443 +187,640 @@ def pol_continuum(data, wavelength, pol_order=3, **polfit_kwargs):
     """
     # Prefix of each table
     prefix = ["346", "437", "580L", "580U", "800U", "860L", "860U"]
     # Path to tables
-    data_path = os.path.join(os.path.dirname(__file__), "..", "input_data", "sky_lines", "ESO-UVES")
+    data_path = os.path.join(
+        os.path.dirname(__file__), "..", "input_data", "sky_lines", "ESO-UVES")
 
-    line_wavelength = np.empty(1)
-    line_fwhm = np.empty(1)
-    line_flux = np.empty(1)
+    # Initialize arrays to store line properties
+    line_wavelength = np.empty(0)
+    line_fwhm = np.empty(0)
+    line_flux = np.empty(0)
 
+    # Read data from each file
     for p in prefix:
         file = os.path.join(data_path, f"gident_{p}.tfits")
         if not os.path.isfile(file):
-            raise NameError(f"File '{file}' could not be found")
+            raise FileNotFoundError(f"File '{file}' could not be found")
+        
         with fits.open(file) as f:
-            wave, fwhm, flux = f[1].data['LAMBDA_AIR'], f[1].data['FWHM'], f[1].data['FLUX']
+            wave = f[1].data['LAMBDA_AIR']
+            fwhm = f[1].data['FWHM']
+            flux = f[1].data['FLUX']
+            
             line_wavelength = np.hstack((line_wavelength, wave))
             line_fwhm = np.hstack((line_fwhm, fwhm))
             line_flux = np.hstack((line_flux, flux))
-    # Sort lines in terms of wavelenth
-    sort_pos = np.argsort(line_wavelength[1:])
-    return line_wavelength[1:][sort_pos], line_fwhm[1:][sort_pos], line_flux[1:][sort_pos]
+
+    # Sort lines by wavelength
+    sort_pos = np.argsort(line_wavelength)
+    return line_wavelength[sort_pos], line_fwhm[sort_pos], line_flux[sort_pos]
 
 # =============================================================================
 # Sky models
 # =============================================================================
-# TODO: convert this class to an ABC
 
 class SkyModel(object):
     """
-    Abstract class of a sky emission model.
+    Abstract class for a sky emission model.
 
     Attributes
     ----------
-    wavelength:
-        TODO
-    intensity:
-        TODO 1D or 2D array, default None
-    variance:
-        TODO
-    verbose:
-        Print messages during the execution.
+    wavelength : np.ndarray
+        1-D array representing the wavelengths of the sky model.
+    intensity : np.ndarray
+        Array representing the intensity of the sky model.
+    variance : np.ndarray
+        Array representing the variance associated with the sky model. 
+        It must have the same dimensions as `intensity`.
+    verbose : bool, optional
+        If True, print messages during execution. Default is True.
+
     Methods
     -------
-    substract
-    substract_PCA
+    subtract(data, variance, axis=-1)
+        Subtracts the sky model from the given data.
+    
+    subtract_pca()
+        Placeholder for PCA subtraction method.
+    
+    vprint(*messages)
+        Print messages if `verbose` is True.
     """
```

Comment:
This certainly makes sense to me, but shouldn't we use `SkySubsCorrection` (or get rid of all the correction classes once and for all ;^D)?